### DISCUSSION TITLE
Issue with Kimai and a HAproxy in Pfsense (OVH)

### DISCUSSION THREAD

**Discussion by nomiskraw** [kimai/kimai#4547] [On-Premise: Installation & Updates] [answered]
### Describe the issue

Hello !
 
I've been trying for 1 week to set up Kimai with a HAproxy reverse proxy which is in a pfsense (2.7.2).

Actually I'm using kimai on another OVH server but on ESXI (a old one) which doesn't have a reverse proxy. My goal is to transfer my kimai to a new OVH server with proxmox on it. My idea was to set up a new kimai (the old one with a .fr and the new one with a .com domain) and put my database back with EasyBackUp.

In HAproxy I've created a frontend and a backend with the right configurations and certificates, but nothing happens...
I put in the Kimai configuration files the trusted proxies with the right address of my reverse but it doesn't change anything.
 
I'd like to point out that I've also set up a nextcloud server, which works (I've changed the .fr to .com) but not the kimai server. Could you please help me ?

### I already tried

- [X] I've read and searched [the documentation](https://www.kimai.org/documentation/).
- [X] I've searched for similar issues in this repository.
- [X] I've searched for similar issues in [the discussions](https://github.com/kimai/kimai/discussions).

### Kimai version

2.8.0

### How do you run Kimai?

Virtual Server or alike

### Which PHP version are you using?

8.1

**Comment by nomiskraw** [comment, marked as answer]
I've just found the solution to my problem: in /etc/nginx/sites-available you must add that it must listen on port 443. Perhaps this solution should be included in the documentation when certbot is not used. 




![Capture d'écran 2024-01-10 173437](https://github.com/kimai/kimai/assets/99745097/1114de0a-0772-4ccd-b717-d83f70abe0f1)

**Comment by nomiskraw** [comment]
I'd like to provide solutions to problems I've encountered.
My connection to the kimai is made with the SAML SSO from Entra (Azur) but I had trouble setting it up. The button on the connection page was displayed correctly and I could enter my identifiers, but once I pressed connect, either I returned to the kimai connection page or it ran and loaded nothing.
 
I noticed that the redirection to kimai was not working properly and that it was in Entra that the kimai application had to be configured.
Redirection after connection is managed in the 'Redirection URI' parameter in Azur Entra.
To access it, go to 
**Applications -> Application registrations -> search for the kimai application you've created -> Authentication -> Web URI redirection.**

Once there, create a new URI with the link to your kimai site.


![Capture d'écran 2024-01-11 101825](https://github.com/kimai/kimai/assets/99745097/ec8ac228-a4c5-4337-8ba3-81ea9acf5420)








After the SSO SAML connection, your kimai site will be redirected !

**Reply by kevinpapst** [reply]
Thanks for sharing 👍
